src/test/java/com/mael/payroll/paymentTypes/HourlyPaymentTest.java

```diff
@@ -0,0 +1,75 @@
+package com.mael.payroll.paymentTypes;
+
+import com.mael.payroll.cards.TimeCard;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.LocalDate;
+
+import static java.time.LocalDate.of;
+import static java.time.Month.JANUARY;
+import static org.junit.Assert.assertEquals;
+
+public class HourlyPaymentTest {
+
+    private HourlyPayment hourlyPayment;
+    private LocalDate friday;
+
+    @Before
+    public void setUp() {
+        hourlyPayment = new HourlyPayment(1000.0);
+        friday = of(2016, JANUARY, 29);
+    }
+
+    @Test
+    public void thePayIsZeroIfNoTimeCard() {
+        assertEquals(0.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test
+    public void thePayIsHoursByRateIfOneTimeCard() {
+        hourlyPayment.addTimeCard(new TimeCard(friday, 8.0));
+        assertEquals(8.0 * 1000.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test
+    public void thePayIsTheSumOfTotalHoursByRateIfTwoTimeCards() {
+        LocalDate monday = of(2016, JANUARY, 25);
+        hourlyPayment.addTimeCard(new TimeCard(monday, 8.0));
+
+        LocalDate tuesday = of(2016, JANUARY, 26);
+        hourlyPayment.addTimeCard(new TimeCard(tuesday, 8.0));
+
+        assertEquals(8.0 * 1000.0 + 8.0 * 1000.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test
+    public void extraHoursArePaidMoreThanNormalHours() {
+        LocalDate monday = of(2016, JANUARY, 25);
+        hourlyPayment.addTimeCard(new TimeCard(monday, 10.0));
+        assertEquals((8.0 + 2.0 * 1.5) * 1000.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test
+    public void paysLessThanEightHours() {
+        LocalDate monday = of(2016, JANUARY, 25);
+        hourlyPayment.addTimeCard(new TimeCard(monday, 5.0));
+        assertEquals(5.0 * 1000.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test
+    public void thePayIsCalculatedWithTheTimeCardsOfCurrentWeekOnly() {
+        LocalDate lastWeek = of(2016, JANUARY, 22);
+        hourlyPayment.addTimeCard(new TimeCard(lastWeek, 8.0));
+
+        LocalDate monday = of(2016, JANUARY, 25);
+        hourlyPayment.addTimeCard(new TimeCard(monday, 8.0));
+
+        assertEquals(8.0 * 1000.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test (expected = HourlyPayment.TimeCardNotFoundException.class)
+    public void throwsExceptionIfNoTimeCardIsFoundForADate() {
+        hourlyPayment.getTimeCard(friday);
```

Comment:
That's an interesting question.
(To which I do not have an immediate solution either ;) )

One thing I could think of is either having a default `ZeroTimeCard` if it is regularly the case that an employee is asked for a time card and you want to make sure you don't have to always check the existence of one.
The main thing that wonders me is, that there is a `throw` statement but it's never caught anywhere. Hence why I'm thinking that it might be overkill at the moment.